text/0000-java-17.md

```diff
@@ -0,0 +1,83 @@
+
+- Start Date: 2023-04-19 
+- RFC PR:
+- FOLIO Issue:
+
+# Java 17
+
+## Summary
+
+From Quesnelia (R2 2023) release on version 17 must be used for JDK and JRE.
+
+## Motivation
+
+### FOLIO Java libraries
+
+FOLIO maintains many Java libraries:
+https://wiki.folio.org/display/REL/pom.xml+dependencies
+
+Some of them are Spring based and have already migrated to Java 17
+because Spring no longer supports Java 11.
+
+As FOLIO still officially supports Java 11 all other FOLIO Java
+libraries must be for Java 11. Java 17 modules can use Java 11
+libraries, but Java 11 modules cannot use Java 17 libraries.
+
+Dropping Java 11 support allows Java libraries to upgrade to Java 17.
+
+### 3rd party Java libraries
+
+Spring no longer supports Java 11. It is likely that other 3rd party
+Java libraries will also drop Java 11 support soon.
+
+### Docker containers
+
+FOLIO DevOps and the FOLIO security team currently need to maintain and
+monitor two JRE containers and two Jenkins JDK containers:
+https://github.com/folio-org/folio-tools/tree/master/folio-java-docker
+https://github.com/folio-org/folio-tools/tree/master/jenkins-slave-docker
+
+Dropping the Java 11 containers reduces the workload.
+
+### Java 11 availability
+
+End of availablility for Java 11 is at least October 2024:
+https://adoptium.net/de/support/
+
+The first flower release that exceeds this date will probably be
+Rocket (R1 2024) because its support period ends around November 2024
+if FOLIO follows a similar cadence to previous releases, see FOLIO support
+policy:
+https://wiki.folio.org/display/PC/PC+Supports+long-term+release+and+regular+release+recommendations
+
+## Detailed Explanation/Design
+
+All Java/JDK/JRE based modules and software libraries are affected.
+
+Releases for Quesnelia and all further flower released must use Java 17
+for build (JDK) and for runtime (JRE).
+
+Orchid and Poppy releases may continue to use Java 11 until these
+flower releases reach their end of life (end of official support period).
+
+## Risks and Drawbacks
+
+The migration of Spring based modules has shown that it's little effort and low risk.
+
+The drawback of this RFC is that it's late, however, a proposal for Orchid
+posted on #tech-council channel in August 2022 got no TC approval.
+
+## Rationale and Alternatives
+
+Alternatives for effective date are Orchid (R1 2023), Poppy (R2 2023),
+and Rocket (R1 2024).
+
+For Orchid it's too late.
+
+Poppy is possible but requires changes to already completed plans
```

Comment:
The planning threshold for Quesnelia is tomorrow, 5th May 2023.

Reply:
> upgrading to the https://wiki.folio.org/display/TC/Officially+Supported+Technologies is always in scope

Where does that policy come from?

Reply:
> Where does the policy you've mentioned come from that "planning threshold for Quesnelia" is due date for this RFC?

I inferred that the `release scope composition` was the point when the scope of a release was decided (and not intended to change further).

I was wrong about that, I agree with you that the `release scope refinement` is a better threshold for this.

> Lazy consensus.

I'm not following how lazy consensus was used to determine the policy that upgrading to officially supported technologies is always included in the scope of work agreed upon. 

I would expect it to be more of a negotiation with the PC and POs when significant changes are made to figure out if those can reasonably fit, especially if we are making that change late in the planning.